nemo/export/trt_llm/converter/model_converter.py

```diff
@@ -83,30 +89,40 @@ def model_to_trtllm_ckpt(
         )
         use_embedding_sharing = True
 
-    weights_dict = convert_model_to_trt_llm_ckpt(
-        model=model,
-        nemo_model_config=nemo_model_config,
-        nemo_export_dir=nemo_export_dir,
-        inference_tp_size=tensor_parallel_size,
-        processes=1,
-        storage_type=dtype,
-        use_parallel_embedding=use_parallel_embedding,
-        decoder_type=decoder_type,
-    )
-
-    world_size = tensor_parallel_size * pipeline_parallel_size
-
-    has_lm_head = "lm_head.weight" in weights_dict
-    if has_lm_head:
-        lm_head_weight = weights_dict["lm_head.weight"]
+    # If the model has been sharded with model parallelism, convert the model in a gpu-distributed manner
+    if use_distributed_convert:
+        weights_dict = dist_model_to_trt_llm_ckpt(
```

Comment:
Is this for the old nemo ckpt type? Old one wasn't called the dist ckpt btw. And we no longer support the old ckpt format.

Reply:
This converts an instantiated nemo model that may be sharded with model parallelism on device, and not a checkpoint

Reply:
You mean this is for the models that are already in the GPU memory?

Reply:
Yeah exactly.

Reply:
It's also why we add new arguments like the vocab size, since we need to unpad the embedding layers, which NeMo pads upon creating the model from the checkpoint.

Reply:
Can you please add this refit option to this script https://github.com/NVIDIA/NeMo/blob/main/scripts/deploy/nlp/deploy_triton.py and this test https://github.com/NVIDIA/NeMo/blob/main/tests/export/nemo_export.py ? Let me take a look how it works.